handlers/doc.go

```diff
@@ -55,10 +54,89 @@ type mesheryPatternResponseWrapper struct {
 type noContentWrapper struct {
 }
 
-// swagger:parameters idGetMesheryPattern idDeleteMesheryPattern
+// swagger:parameters idGetMesheryPattern idDeleteMesheryPattern idGetSinglePerformanceProfile idDeletePerformanceProfile idGETProfileResults
 type IDParameterWrapper struct {
-	// id to fetch the item
+	// id for a specific
 	// in: path
 	// required: true
-	ID uuid.UUID `json:"id"`
+	ID strfmt.UUID `json:"id"`
+}
+
+// Returns all the performance profiles
+// swagger:response performanceProfilesResponseWrapper
+type performanceProfilesResponseWrapper struct {
+	// in: body
+	Body models.PerformanceProfilesAPIResponse
+}
+
+// Returns all performance results
+// swagger:response performanceResultsResponseWrapper
+type performanceResultsResponseWrapper struct {
+	// in: body
+	Body models.PerformanceResultsAPIResponse
+}
+
+// Returns a single performance profile
+// swagger:response performanceProfileResponseWrapper
+type performanceProfileResponseWrapper struct {
+	// in: body
+	Body models.PerformanceProfile
+}
+
+// Save a performance profile
+// swagger:parameters idSavePerformanceProfile
+type performanceProfileParameterWrapper struct {
+	// in: body
+	// name of performance profile
+	// required: true
+	Name string `json:"name,omitempty"`
+	// array of load generators
+	// required: true
+	LoadGenerators []string `json:"load_generators,omitempty" gorm:"type:text[]"`
+	// array of urls of performance results
+	// required: true
+	Endpoints []string `json:"endpoints,omitempty" gorm:"type:text[]"`
+	// service mesh for performance tests
+	// required: true
+	ServiceMesh string `json:"service_mesh,omitempty"`
+	// number of concurrent requests
+	// required: true
+	ConcurrentRequest int `json:"concurrent_request,omitempty"`
+	// qps in integer
+	// required: true
+	QPS int `json:"qps,omitempty"`
+	// duration of tests e.g. 30s
+	// required: true
+	Duration string `json:"duration,omitempty"`
+}
```

Comment:
Can all this be defined in an existing model instead of having to redefine it? IDK if this is indeed the right way. If it is then okay.

Reply:
NVM, I think that is the way to go

Reply:
You're right, I'll shift these in models, next time as this file will start to get clustered if I'll keep declaring these parameters here.

Reply:
Is this being imported from the https://github.com/layer5io/service-mesh-performance project protos?

Reply:
Nope, I'm not importing anything from this package.

Reply:
@piyushsingariya some of the models can be reused from in here - https://github.com/layer5io/service-mesh-performance/blob/master/spec/service_mesh_performance.pb.go

Reply:
@kumarabd I haven't declared any structs that could have been imported from here. I just declared some query parameter structs. I'll look into this for future reference for writing response structs.

Reply:
I was wondering about the use of wrapper here, makes sense to me. Good to go.